File: 110/balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def __init__(self):
        self.balanced = True

    def isBalanced(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        self.dfs(root)
        return self.balanced

    def dfs(self, root):
        """
        :type root: TreeNode
        :rtype depth: Number
        """
        if None == root:
            return
        left_height = self.dfs(root.left) if root.left else 0
        right_height = self.dfs(root.right) if root.right else 0

        if abs(left_height - right_height) > 1:
            self.balanced = False

        return max(left_height, right_height) + 1

File: 110/balanced-binary-tree.go
import "math"

func isBalanced(root *TreeNode) bool {
	balanced := true
	func dfs(root *TreeNode) int {
		if nil == root {
			return
		}
		left_height = root.Left != nil ? dfs(root.Left) : 0
		right_height = root.Right != nil ? dfs(root.Right) : 0

		if math.Abs(left_height - right_height) > 1 {
			balanced = false
		}

		return math.Max(left_height, right_height) + 1
	}
	dfs(root)
	return balanced
}